homeassistant/util/__init__.py

```diff
@@ -260,25 +285,21 @@ def wrapper(*args, **kwargs):
             else:
                 host = args[0] if args else wrapper
 
-            if not hasattr(host, '_throttle_lock'):
-                host._throttle_lock = threading.Lock()
+            if not hasattr(host, '_throttle'):
+                host._throttle = {}
 
-            if not host._throttle_lock.acquire(False):
-                return None
+            if not id(self) in host._throttle:
+                host._throttle[id(self)] = ThrottleItem()
```

Comment:
Namedtuples and datetime objects are both immutable.

I actually think it would improve the readability a lot because all logic is contained within 20 lines. There is no need to scroll.